wsd/DocumentBroker.cpp

```diff
@@ -1893,6 +1893,18 @@ void DocumentBroker::handleSaveResponse(const std::shared_ptr<ClientSession>& se
                                                     << DocumentState::name(_docState.activity())
                                                     << ") in " << _saveManager.lastSaveDuration());
 
+    // Update the save timeout, so we don't give up too soon, with 50% headroom.
+    const auto minTimeoutSecs =
+        std::chrono::ceil<std::chrono::seconds>(_saveManager.lastSaveDuration() * 1.5);
```

Comment:
I really don't like this; where does 1.5 come from ? documents can grow significantly between saves - what is the outcome then ? If this method: setSavingTimeout

wsd/DocumentBroker.hpp-        /// Set the maximum time to wait for saving to finish.

Should be called as you suggest on IRC - setSaveDebounceTimeout or something - and just doesn't allow another save in that window - then perhaps it's fine - but if we go around disabling background save rabidly when people eg. copy/paste lots of data into their blank spreadsheet - we just lost a big win - in it's most useful use-case.

Reply:
I excluded it.

This isn't new, though. The idea is to block the possibility of overlapping saves, while eventually giving up and resume accepting save commands. Perhaps we no longer need it now that we have the watchdog, but, like the watchdog, there has to be _some_ timeout, somewhere. If 50% over last save response is not reasonable (with minimum of 5 seconds), we can make this more liberal and even configurable, I guess.